node-memory-considerate-loading/server.js

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the 'License');
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const express = require('express');
+// const bodyParser = require('body-parser');
+const path = require('path');
+const cors = require('cors');
+const request = require('request');
+
+const app = express();
+app.disable('x-powered-by');
+app.use(cors());
+
+app.use(express.static(path.join(__dirname, 'public')));
+app.set('views', __dirname + '/public');
+app.engine('html', require('ejs').renderFile);
+app.set('view engine', 'html');
+
+app.get('/ping', (req, res) => {
+  res.send('pong');
+});
+
+app.get('/memory-considerate-image', (req, res) => {
+  // TODO: As this is a demo, I think it should be easy enough to change these numbers as needed in the future.
+  const MEMORY_LIMIT = 4;
+  const deviceMemory = req.headers['device-memory'];
+  console.log('[server memory-considerate-image request] Device Memory => ', deviceMemory);
+  const url = deviceMemory < MEMORY_LIMIT ?
+    'https://cdn.glitch.com/8d7fb7f0-a9be-4a8c-96c7-8af286af487e%2Fmin-res.jpg?v=1562842586912' :
```

Comment:
We probably shouldn't rely on Glitch-hosted assets (as much as we do right now). Could we locally host these in the project and reference those instead? That way if the upstream Glitch goes down, everything still works... :)